lib/ecto/changeset/relation.ex

```diff
@@ -252,6 +251,27 @@ defmodule Ecto.Changeset.Relation do
     Changeset.change(current, changes) |> put_new_action(:update)
   end
 
+  @doc """
+  TODO
+  """
+  def on_replace(%{on_replace: :mark_as_invalid}, changeset_or_model) do
+    %{Changeset.change(changeset_or_model) | valid?: false}
```

Comment:
We would always add the error to the relation field in the parent changeset. I think this is the best way to go because if we support `put_change(changeset, :relation, on_replace: :mark_as_invalid)`, i would expect the field to be marked as invalid right there. So we should change how `put_change` works slightly then for relations.